### PR TITLE
Added cup of molten gold action

### PR DESCRIPTION
**Pull request by EPuncker**
Added on 9.1 version

https://tibia.fandom.com/wiki/Cup_of_Molten_Gold

script by me
tested

### CODE REVIEW

File: data/scripts/actions/others/cup_of_molten_gold.lua [hunk 1]
```diff
@@ -0,0 +1,25 @@
+local cupOfMoltenGold = Action()
+
+function cupOfMoltenGold.onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	local targetId = target.itemid
+	if not table.contains({2700, 21428}, targetId) then -- fir tree or fir cone
```
Comment (DSpeichert) [inline data/scripts/actions/others/cup_of_molten_gold.lua:5]: is it a typo? fire tree / fire cone?

File: data/scripts/actions/others/cup_of_molten_gold.lua [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		return false
+	end
+	if math.random(100) <= 10 then
+		if targetId == 21428 then
+			item:transform(13539)
```
Comment (DSpeichert) [inline data/scripts/actions/others/cup_of_molten_gold.lua:10]: please annotate with item name

File: data/scripts/actions/others/cup_of_molten_gold.lua [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		else
+			player:addItem(13539, 1)
```
Comment (DSpeichert) [inline data/scripts/actions/others/cup_of_molten_gold.lua:12]: please annotate with item name here as well

File: data/scripts/actions/others/cup_of_molten_gold.lua [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		end
+	else
+		player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Drizzling all over a fir cone you picked from the tree, the molten gold only covers about half of it - not enough.")
+		if targetId == 21428 then
```
Comment (DSpeichert) [inline data/scripts/actions/others/cup_of_molten_gold.lua:16]: please annotate with item name

File: data/scripts/actions/others/cup_of_molten_gold.lua (hunk 1 above)
Comment (M4g1ch) [reply data/scripts/actions/others/cup_of_molten_gold.lua:5]: No, it's not a typo

Comment (EPuncker) [reply data/scripts/actions/others/cup_of_molten_gold.lua:5]: not a typo but the tree names 😆

**Pull request merged**